app/controllers/wholesalers_controller.rb

```diff
@@ -42,21 +46,28 @@ def edit
 
   def update
     @wholesaler = Wholesaler.find(params[:id])
-    if @wholesaler.update(wholesaler_params)
+    if wrong_user
+      redirect_to '/customers/sign_in'
+    elsif @wholesaler.update(wholesaler_params)
       flash[:success] = 'Updated!'
-      flash[:success] = 'Approved!' if params[:is_approved] == 'true'
-      flash[:success] = 'Rejected!' if params[:is_rejected] == 'true'
+      # flash[:success] = 'Approved!' if params[:is_approved] == 'true'
+      # flash[:success] = 'Rejected!' if params[:is_rejected] == 'true'
```

Comment:
Just nitpicking, but a different flash status may be appropriate for a rejection message. Maybe `flash[:warning]`, `flash[:alert]`